File: convert_multi_answer.py
import re
import json

# Read the questions file
with open('lib/questions.ts', 'r', encoding='utf-8') as f:
    content = f.read()

# Pattern to match question objects
question_pattern = r'\{\s*id:\s*(["\']dumps-\d+["\']),\s*text:\s*(["\'].*?["\'])\s*,\s*topic:\s*(["\'].*?["\'])\s*,\s*options:\s*\[(.*?)\],\s*correctAnswer:\s*([^,]+),\s*difficulty:\s*(["\'][^"\']*["\'])\s*,\s*explanation:\s*(["\'].*?["\'])'

# Find all multi-answer questions
matches = list(re.finditer(question_pattern, content, re.DOTALL))

multi_answer_info = []

for match in matches:
    question_id = match.group(1).strip("'\"")
    text = match.group(2).strip("'\"")
    explanation = match.group(7).strip("'\"")
    
    # Check if this is a multi-answer question
    if 'Choose 2 answers' in text or 'Choose 3 answers' in text:
        correct_count = 2 if 'Choose 2 answers' in text else 3
        multi_answer_info.append({
            'id': question_id,
            'text': text[:100] + '...',
            'explanation': explanation[:150] + '...',
            'correct_count': correct_count,
            'match_start': match.start(),
            'match_end': match.end(),
            'full_match': match.group(0)
        })

print(f"Found {len(multi_answer_info)} multi-answer questions")
print("\nFirst 10 multi-answer questions:")
for i, info in enumerate(multi_answer_info[:10]):
    print(f"{i+1}. {info['id']}: {info['text']}")
    print(f"   Needs {info['correct_count']} answers")
    print(f"   Explanation: {info['explanation']}\n")

# Save info for manual review
with open('multi_answer_questions.json', 'w') as f:
    json.dump(multi_answer_info, f, indent=2)

print(f"\nTotal: {len(multi_answer_info)} questions need conversion")
print("Saved question details to multi_answer_questions.json")

File: convert_multi_answer.js
const fs = require('fs');
const path = require('path');

// Read the questions file
const filePath = path.join(__dirname, 'lib', 'questions.ts');
const content = fs.readFileSync(filePath, 'utf-8');

// Split by question objects
const lines = content.split('\n');

const multiAnswerQuestions = [];

let i = 0;
while (i < lines.length) {
  if (lines[i].includes('Choose 2 answers') || lines[i].includes('Choose 3 answers')) {
    // Found a multi-answer question
    let questionStart = i;
    
    // Search backwards to find the id
    let id = '';
    for (let j = i; j >= Math.max(0, i - 20); j--) {
      const idMatch = lines[j].match(/id:\s*'(dumps-\d+)'/);
      if (idMatch) {
        id = idMatch[1];
        break;
      }
    }
    
    // Extract the "Choose X answers" pattern
    const chooseMatch = lines[i].match(/Choose (\d+) answers/);
    const correctCount = chooseMatch ? parseInt(chooseMatch[1]) : 2;
    
    // Search forward to find explanation
    let explanation = '';
    for (let j = i; j < Math.min(lines.length, i + 15); j++) {
      const explMatch = lines[j].match(/explanation:\s*'([^']+)'/);
      if (explMatch) {
        explanation = explMatch[1];
        break;
      }
    }
    
    multiAnswerQuestions.push({
      id,
      text: lines[i].substring(0, 80) + '...',
      correctCount,
      explanation: explanation.substring(0, 120) + '...',
      line: i + 1
    });
  }
  i++;
}

console.log(`Found ${multiAnswerQuestions.length} multi-answer questions\n`);
console.log('First 15 questions:');
multiAnswerQuestions.slice(0, 15).forEach((q, idx) => {
  console.log(`${idx + 1}. ${q.id} (line ${q.line}): needs ${q.correctCount} answers`);
  console.log(`   ${q.explanation}\n`);
});

fs.writeFileSync('multi_answer_list.json', JSON.stringify(multiAnswerQuestions, null, 2));
console.log(`\nTotal: ${multiAnswerQuestions.length} questions found`);
